frontend/src/components/core/Block/Block.tsx

```diff
@@ -426,10 +437,13 @@ class Block extends PureComponent<Props> {
   public render = (): ReactNode => {
     if (this.props.deltaBlock && this.props.deltaBlock.horizontal) {
       // Create a horizontal block as the parent for columns
-      // For now, all children are column blocks, so we can ignore `width`.
+      // For now, all children are column blocks, and `width` is driven by
+      // the total number of columns available.
       return (
-        <div className="stBlock-horiz" style={{ display: "flex", gap: "8px" }}>
-          {this.renderElements(0)}
+        <div className="stBlock-horiz">
+          {this.renderElements(
+            this.props.deltaBlock.horizontal.totalColumns || 0
```

Comment:
I _think_ we can remove `|| 0` since that's what totalColumns would default to if it's not filled in somehow? And looking at the code path it should also always be filled in.

Reply:
This was just to make typescript happy because i didn't want to do a null check lol. Is there a way in protobuf to make it not nullable?

Reply:
Oh, I see. In my head, accessing a nonexistent protobuf field was supposed to just work and return the default value for the type (e.g. 0 for numbers, False for bools, '' for strings), but seems like that's not how it actually works.

Let's just leave it as-is!